src/converse-sendbutton.js

```diff
@@ -0,0 +1,94 @@
+// Converse.js (A browser based XMPP chat client)
+// http://conversejs.org
+//
+// Copyright (c) 2012-2017, Jan-Carel Brand <jc@opkode.com>
+// Licensed under the Mozilla Public License (MPLv2)
+//
+/*global Backbone */
+
+/* This plugin is to add an option of to have send button in chat box */
+
+(function(root, factory) {
+    define([
+        "converse-core",
+        "converse-chatview",
+        "converse-muc",
+        "converse-headline"
+    ], factory);
+}(this, function(converse) {
+    "use strict";
+    var $ = converse.env.jQuery;
+
+    converse.plugins.add('converse-sendbutton', {
+
+        overrides: {
+            // Overrides mentioned here will be picked up by converse.js's
+            // plugin architecture they will replace existing methods on the
+            // relevant objects or classes.
+            //
+            // New functions which don't exist yet can also be added.
+
+            ChatBoxView: {
+                events: {
+                    'click .send-button': 'onSendButtonClicked'
+                },
+
+                onSendButtonClicked: function(ev) {
+                    /* Event handler for when a send button is clicked in a chat box textarea.
+                     */
+                    var _converse = this.__super__._converse;
+                    ev.preventDefault();
+                    var textarea = $(ev.target).siblings('.chat-textarea')[0],
```

Comment:
@jcbrand  I think should we should make a helper library for this type of most commonly used functions so that we do not write them again and again. what say?